tests/foreman/cli/test_realm.py

```diff
@@ -35,6 +36,14 @@ def _make_proxy(options=None):
     capsule_cleanup(proxy['id'])
 
 
+@pytest.fixture(scope='function')
+def _make_realm(options=None):
```

Comment:
This is not a proper pytest fixture pattern - the fixture does not take arguments, and is not called directly by the test.

This is calling `make_realm` with no options, and yielding the entity instance returned by that function.  The test function is then calling `_make_realm` again, with options passed this time, creating the realm entity twice.

Parameters in the function signature are themselves other fixture names, that pytest will resolve and execute.

If you want to add realm cleanup, you can do so with `request.addfinalizer(realm_cleanup(realm['id'])` within the test functions, after the realm has been created.  Include the `request` fixture in the test function parameters.

There is a valid fixture use pattern where the fixture returns a function object, which can then be called within the test, but that's not necessary here and is best reserved for more complex usage.

Reply:
updated the pr.